### ISSUE TITLE
Not taking input on android sdk 32 and returning false error on speech start

### ISSUE THREAD

**Issue by ayushgupta-1** [react-native-voice/voice#429]
I am working on android 12 and this library is working fine for iOS but on android, it is not taking any input from microphone, returns false on speech start.

Also no error anywhere, mic turns on and turns off so nothing unusual there.

const onSpeechStartHandler = e => {
console. log ('start handler=»', e);
};

Output : Start handiers=» {"error": false}

does anyone know any solution for the same ?

**Comment by WinKoKoNaing** [comment]
I am facing same issue, have you found solution?

**Comment by ayushgupta-1** [comment]
Not yet, seems this library is not taking the android default speech engine and thus giving the error.

**Comment by khadeshyam** [comment]
can you show the output of react-native-info in root project on our development environment?

**Comment by ayushgupta-1** [comment]
I didn't get about react-native-info ?? please check stack overflow, I have posted as a question there with screenshots and code. Thanks.

**Comment by khadeshyam** [comment]
To help troubleshoot the issue, please run react-native-info in the root project directory and share the output. 
This will help identify any compatibility issues between the library and your React Native version. 
And please try using react-native-community/voice package instead of react-native-voice/voice it's depreciated and no longer maintained

**Comment by ayushgupta-1** [comment]
<img width="781" alt="Screenshot 2023-04-02 at 11 58 24 AM" src="https://user-images.githubusercontent.com/84156356/229336425-25bdeaf9-746b-4cc4-964f-df9c09965932.png">

I guess you are asking about this. Check my query here : [Stack overflow](https://stackoverflow.com/questions/75742318/react-native-voice-speech-to-text-not-working-in-android)

**Comment by khadeshyam** [comment]
I think you have not setup android development environment on your computer properly.visit this [link](https://reactnative.dev/docs/next/environment-setup?os=macos) to setup android development environment on your computer.

**Comment by ayushgupta-1** [comment]
After running react-native info
![Screenshot 2023-04-02 at 1 13 59 PM](https://user-images.githubusercontent.com/84156356/229339680-a162337c-3c60-4a93-8112-cffaa9c45971.png)

After running npx react-native doctor
![Screenshot 2023-04-02 at 1 14 45 PM](https://user-images.githubusercontent.com/84156356/229339705-61b82260-332c-441d-9ca9-9ce89a498e9e.png)

I have everything and still it is not working, have you seen my stack overflow question @khadeshyam ? 

The start speech handler is returning false
LOG : start handler=» {"error": false}

**Comment by khadeshyam** [comment]
yes you are correct react-native-community/voice is depreciated but try using @react-native-community/voice
it does work on my android phone

**Comment by khadeshyam** [comment]
Did you try to setup android linking of the project manually?

**Comment by ayushgupta-1** [comment]
Yes, I have tried linking manually before and was getting error that library added or linked twice. 

react-native-community/voice is nowhere to be found. So, I can't take risk of adding a library that I already know has deprecated as the project I am working on is not a personal project and at least iOS is working fine for now.

Also I believe the issue is something with the native code.

**Comment by trieulongben** [comment]
Hello, in my case i do this solution:
node_modules-> react-native-voice/voice -> android -> open androidManifest and add this block

<queries>
        <intent>
            <action android:name="android.speech.RecognitionService" />
        </intent>
    </queries>

I tested on Android 13 device, try this ✌️

**Comment by wallinex** [comment]
It tested on android 11, 12, and 13. I added the details to the manifest:
`
<queries>
        <intent>
            <action android:name="android.speech.RecognitionService" />
        </intent>
    </queries>`
    
    Still not working. 
    I use this version: "@react-native-voice/voice": "^3.2.4",
    
    It logs that it's recording
    ` LOG  State: {"end": "", "error": "", "isRecording": true, "partialResults": [], "pitch": 0.6399998664855957, "recognized": "", "results": [], "started": "√"}`
    
    I get this when I run npx expo-doctor
    `✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check native tooling versions
✔ Check if the project meets version requirements for submission to app stores
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✔ Check for app config fields that may not be synced in a non-CNG project
✔ Check npm/ yarn versions
✔ Check for issues with metro config
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check that packages match versions required by installed Expo SDK
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check that native modules use compatible support package versions for installed Expo SDK

Didn't find any issues with the project!`

Anyone else having these issues?

**Comment by williamjayjay** [comment]
> Hello, in my case i do this solution: node_modules-> react-native-voice/voice -> android -> open androidManifest and add this block
> 
> ```
> 
> <queries>
>         <intent>
>             <action android:name="android.speech.RecognitionService" />
>         </intent>
>     </queries>
> ```
> 
> I tested on Android 13 device, try this ✌️

this work to me too! thanks
